controllers/client/payment.controller.js

```diff
@@ -230,10 +231,14 @@ export const createOrderIdForPendingPaymentsV2 = async (req, res) => {
   if (totalAmount > 0) {
     const order = await generateOrderId(totalAmount);
     const validTransactionIds = transactions
-      .filter((t) =>
-        bookingCategoryMap[t.bookingid].includes(getBookingType(t))
-      )
-      .map((t) => t.id);
+      .filter((transaction) => {
+        const categories = bookingCategoryMap[transaction.bookingid];
+        const bookingType = getBookingType(transaction);
+        return (
+          bookingType != TYPE_FOOD || categories.includes(transaction.category)
+        );
+      })
```

Comment:
I think it's still risky as it duplicates the logic of filtering transactions in two places, on line 225 and 238. looking at the history of this file, I believe that's what caused this bug when we tried to fix this 3 weeks ago.